src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -620,3 +636,245 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               runTimer() {
+                  this.timerId = engine.beginTimer(20, function() {
+                     engine.setValue('[Test]', 'co', this.globVar);
+                     this.globVar++;
+                     engine.setValue('[Test]', 'coTimerId', this.timerId + 10);
+                  }.bind(this), true);            
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
+            var MyMapping = new MyClass();
+            MyMapping.runTimer();)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(timerId + 10, coTimerId->get());
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    EXPECT_TRUE(evaluateAndAssert("engine.setValue('[Test]', 'co', MyMapping.globVar);"));
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(8.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_singleShotTimerArrowFunctionInClass) {
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               runTimer() {
+                  this.timerId = engine.beginTimer(20, ()=> {
+                     engine.setValue('[Test]', 'co', this.globVar);
+                     this.globVar++;
+                     engine.setValue('[Test]', 'coTimerId', this.timerId + 10);
+                  }, true);            
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
+            var MyMapping = new MyClass();
+            MyMapping.runTimer();)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(timerId + 10, coTimerId->get());
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    EXPECT_TRUE(evaluateAndAssert("engine.setValue('[Test]', 'co', MyMapping.globVar);"));
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(8.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_repeatedTimerArrowFunctionCallInClass) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               stopTimer() {
```

Comment:
It would make sense to explicitly check whether `this` has actually the value we expect here (using a hypothetical `assertThat` function):

```suggestion
               stopTimer() {
                   assertThat(this instanceof MyClass);
```